rest/view_api.go

```diff
@@ -29,6 +30,11 @@ func (h *handler) handleGetDesignDoc() error {
 		result = db.Body{"filters": db.Body{"bychannel": filter}}
 	} else {
 		if err := h.db.GetDesignDoc(ddocID, &result); err != nil {
+			// GoCB doesn't provide an easy way to distinguish what the cause of the error was, so
+			// resort to a string pattern match for "not_found" and propagate a 404 error in that case.
+			if strings.Contains(err.Error(), "not_found") {
+				return base.HTTPErrorf(404, "%v","missing")
+			}
```

Comment:
The gocb error introspection will be easier to maintain if we push this error checking down into CouchbaseBucketGoCB.GetDDoc, and returned a new ErrNotFound SGError that we could check for here.

Reply:
Good catch, fixed.

Reply:
Actually, some more work needed to create and return `ErrNotFound `